storage/client_replica_test.go

```diff
@@ -884,3 +885,60 @@ func TestErrorHandlingForNonKVCommand(t *testing.T) {
 		t.Fatalf("expected error %q, got: %s", "injected error", pErr)
 	}
 }
+
+func newTransaction(name string, baseKey roachpb.Key, userPriority roachpb.UserPriority,
+	isolation enginepb.IsolationType, clock *hlc.Clock) *roachpb.Transaction {
+	var offset int64
+	var now hlc.Timestamp
+	if clock != nil {
+		offset = clock.MaxOffset().Nanoseconds()
+		now = clock.Now()
+	}
+	return roachpb.NewTransaction(name, baseKey, userPriority, isolation, now, offset)
+}
+
+// Test that a duplicate BeginTransaction results in a TransactionRetryError, as
+// such recognizing that it's likely the result of the batch being retried by
+// DistSender.
+func TestDuplicateBeginTransaction(t *testing.T) {
```

Comment:
This test is unnecessarily high up in the stack. Why not put it close to `TestReplayProtection` in `replica_test.go`?